gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/format/PalantirJavaFormatStep.java

```diff
@@ -0,0 +1,119 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.plugins.format;
+
+import com.diffplug.spotless.FormatterFunc;
+import com.diffplug.spotless.FormatterStep;
+import com.diffplug.spotless.JarState;
+import com.diffplug.spotless.Provisioner;
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.lang.reflect.Method;
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.gradle.api.artifacts.Configuration;
+import org.jetbrains.annotations.NotNull;
+
+public final class PalantirJavaFormatStep {
+
+    private PalantirJavaFormatStep() {}
+
+    private static final String NAME = "palantir-java-format";
+
+    private static final String FORMATTER_CLASS = "com.palantir.javaformat.java.Formatter";
+    private static final String FORMATTER_CREATE_METHOD = "createFormatter";
+
+    private static final String FORMATTER_METHOD = "formatSourceAndFixImports";
+
+    private static final String OPTIONS_CLASS = "com.palantir.javaformat.java.JavaFormatterOptions";
+    private static final String OPTIONS_BUILDER_METHOD = "builder";
+    private static final String OPTIONS_BUILDER_CLASS = "com.palantir.javaformat.java.JavaFormatterOptions$Builder";
+    private static final String OPTIONS_BUILDER_STYLE_METHOD = "style";
+    private static final String OPTIONS_BUILDER_BUILD_METHOD = "build";
+    private static final String OPTIONS_Style = "com.palantir.javaformat.java.JavaFormatterOptions$Style";
+
+    /** Creates a step which formats everything - code, import order, and unused imports. */
+    public static FormatterStep create(Configuration palantirJavaFormat) {
+
+        return FormatterStep.createLazy(
+                NAME,
+                () -> new State(
+                        palantirJavaFormat.getAllDependencies().stream().map(Object::toString).collect(
+                                Collectors.toList()),
+                        new ConfigurationBackedProvisioner(palantirJavaFormat)),
+                State::createFormat);
+    }
+
+    static final class State implements Serializable {
+        private static final long serialVersionUID = 1L;
+
+        /** Kept for state serialization purposes. */
+        @SuppressWarnings("unused")
+        private final String stepName = NAME;
+        /** The jar that contains the palantir-java-format implementation. */
+        private final JarState jarState;
+
+        State(Collection<String> deps, Provisioner provisioner) throws IOException {
+            this.jarState = JarState.from(deps, provisioner);
+        }
+
+        @SuppressWarnings({"unchecked", "rawtypes"})
+        FormatterFunc createFormat() throws Exception {
+            ClassLoader classLoader = jarState.getClassLoader();
+
+            Class<?> optionsClass = classLoader.loadClass(OPTIONS_CLASS);
```

Comment:
after https://github.com/palantir/palantir-java-format/pull/47, everything down from here becomes super simple and doesn't need reflection anymore: just service load the FormatterService from the classLoader.